deploy-board/deploy_board/webapp/cluster_view.py

```diff
@@ -188,14 +195,19 @@ def post(self, request, name, stage):
 
             log.info("Create Capacity in the provider")
             clusters_helper.create_cluster(request, cluster_name, cluster_info)
-
-            return HttpResponse("{}", content_type="application/json")
         except NotAuthorizedException as e:
             log.error("Have an NotAuthorizedException error {}".format(e))
-            return HttpResponse(e, status=403, content_type="application/json")
+            ret = 403
         except Exception as e:
             log.error("Have an error {}", e)
-            return HttpResponse(e, status=500, content_type="application/json")
+            ret = 500
+        finally:
+            if ret == 200:
+                return HttpResponse("{}", content_type="application/json")
+            else:
+                environs_helper.remove_env_capacity(
```

Comment:
will this remove the cluster?  can you make sure this command will rollback the previous action:  environs_helper.update_env_basic_config and environs_helper.add_env_capacity? 

also, what if the error happen in environs_helper.update_env_basic_config or environs_helper.add_env_capacity, will this call failed unexpectedly.

Reply:
looks like these block of code is exactly from previous method, can we consolidate them into a common function?

Reply:
Yes, environs_helper.remove_env_capacity also removes the clusterName in environs.
If error happens during db operations, then we definitely will get an error on UI. After that, everything will be consistent.
For basic and advanced page, the post method has a little bit difference.